backend/LexBoxApi/GraphQL/ProjectMutations.cs

```diff
@@ -167,6 +166,37 @@ public record BulkAddProjectMembersResult(List<UserProjectRole> AddedMembers, Li
         return new BulkAddProjectMembersResult(AddedMembers, CreatedMembers, ExistingMembers);
     }
 
+    public static (string name, string? email, string? username) ExtractNameAndAddressFromUsernameOrEmail(string usernameOrEmail)
+    {
+        var isEmailAddress = usernameOrEmail.Contains('@');
+        string name;
+        string? email;
+        string? username;
+        if (isEmailAddress)
+        {
+            try
+            {
+                var parsed = new MailAddress(usernameOrEmail);
+                email = parsed.Address;
+                username = null;
+                name = parsed.DisplayName;
+                if (string.IsNullOrEmpty(name)) name = email.Split('@')[0];
+            }
+            catch (FormatException)
+            {
+                // FormatException message from .NET talks about mail headers, which is confusing here
+                throw new InvalidEmailException("Invalid email address", usernameOrEmail);
+            }
```

Comment:
Can you actually put this in its own method and use it to validate the email address in `AddProjectMember` as well.